ui/src/src/common/uicomponents/ConfirmButton.js

```diff
@@ -0,0 +1,53 @@
+import SubComponent from "../SubComponent";
+import React from "react";
+
+export class ConfirmButton extends SubComponent {
+    constructor(props) {
+        super(props);
+        this.state = {
```

Comment:
It looks like this needs a second thought as I see a number of counter-patterns:
1. Storing event handlers in a component state
2. passing props into the state.
3. A stateful button does not look intuitive, so probably there's a better solution for that.